crates/router/src/core/payments/helpers.rs

```diff
@@ -1345,7 +1454,26 @@ pub async fn make_pm_data<'a, F: Clone, R, Ctx: PaymentMethodRetrieve>(
                         },
                     ))?;
 
-                Some(key)
+                let token_data_result = token_data_string
+                    .clone()
+                    .parse_struct("HyperswitchTokenData")
+                    .change_context(errors::ApiErrorResponse::InternalServerError)
+                    .attach_printable("failed to deserialize hyperswitch token data");
```

Comment:
So token is being serialized as a `PaymentTokenData` and expected to be parsed as `HyperswitchTokenData`?
So is the other struct for old entries in redis?